### PR TITLE
Migrate to new dataloaders

### PR DESCRIPTION
**Pull request by TroyKomodo**
## Proposed changes

This commit migrates to the new dataloaders and also takes the work to move the old db struct style to the new style. Syncing with the video side.

Remaining work would be to update the protobuf files for events to stop using strings and make use of the Ulid proto.

## Types of changes

What types of changes does your code introduce to Scuffle?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/ScuffleTV/scuffle/blob/main/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: platform/api/src/api/v1/request_context.rs [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
+    database::{Role, RolePermission, Session, User},
+    global::GlobalState,
+};
 
 #[derive(Clone)]
 pub struct AuthData {
-    pub session: session::Model,
-    pub user_roles: Vec<role::Model>,
-    pub user_permissions: role::Permission,
+    pub user: User,
```
Comment (lennartkloock) [inline platform/api/src/api/v1/request_context.rs:14]: Why did you add user here?

Comment (TroyKomodo) [reply platform/api/src/api/v1/request_context.rs:14]: Because we had it as a byproduct. I wasnt sure if we needed it

Comment (lennartkloock) [reply platform/api/src/api/v1/request_context.rs:14]: I excluded it on purpose because it could have outdated information after a few requests

Comment (TroyKomodo) [reply platform/api/src/api/v1/request_context.rs:14]: Ah, okay.
